src/document/DocumentAnnotator.ts

```diff
@@ -64,6 +60,19 @@ export default class DocumentAnnotator extends BaseAnnotator {
         // Lazily instantiate managers as pages are added or re-rendered
         if (managers.size === 0) {
             if (this.isFeatureEnabled('highlightText')) {
+                const textLayer = pageEl.querySelector('.textLayer') as HTMLElement;
+
+                if (textLayer) {
+                    managers.add(
+                        new HighlightCreatorManager({
+                            getSelection,
+                            referenceEl: textLayer,
+                            selectionChangeDelay: TEXT_LAYER_ENHANCEMENT,
+                            store: this.store,
+                        }),
+                    );
+                }
+
                 managers.add(new HighlightManager({ location: pageNumber, referenceEl: pageReferenceEl }));
```

Comment:
Should this manager be moved into the `if (textLayer)` condition, as well? Or do we want to render existing highlights for those without download permissions?

Reply:
Good question. I think we still want to leave it outside because, as you said, we still want to render existing highlights -- they also show up in the activity feed still (even without download permissions)